### PR TITLE
NA

### PR DESCRIPTION
**Pull request by JiatangDong**
N/A

### CODE REVIEW

**Review by potatowagon — COMMENTED** [review]
Reviewed by Navi (Tater Thoughts Bobblehead) on behalf of @potatowagon.

**Assessment: LGTM** ✅ — Playground-only feature (GitHub code permalink embeds). No core library changes, no www impact.

**What I checked:**

1. **Node implementation (GitHubCodeNode.tsx, 502 lines):** Well-structured DecoratorBlockNode subclass. Proper `importJSON`/`exportJSON` round-trip with `updateFromJSON`. `exportDOM` uses data attributes. `getTextContent()` returns the URL (good for copy-paste). Uses `LexicalNestedComposer` with `createEditor` for syntax highlighting via `@lexical/code-prism` — readonly nested editor, non-editable. Clean React component with AbortController cleanup, progressive line reveal (show more/collapse), refresh button. All state management is local.

2. **URL parsing (AutoEmbedPlugin/index.tsx):** Supports blob, blame, gist, and raw.githubusercontent.com patterns. Language detection from file extension via `LANGUAGE_BY_EXTENSION` map. Regex patterns are well-scoped and tested. The `isGitHubCodeMatchResult` type guard is clean.

3. **Extension registration (GitHubCodeExtension/index.ts):** Uses `defineExtension` + `$insertNodeToNearestRoot` + `COMMAND_PRIORITY_EDITOR`. Standard pattern consistent with FigmaExtension, TwitterExtension, etc.

4. **Test coverage:** 272-line unit test (creation, type checking, serialization round-trip, DOM export, cloning) + 185-line URL parsing test (blob, blame, gist, raw, invalid URLs). Comprehensive edge case coverage.

5. **CSS (155 lines):** GitHub-style embed UI. No global selector leaks — all scoped under `.github-code-*`. Responsive grid layout for line numbers + code.

6. **www compat:** No impact — purely additive playground feature with no core library API changes, no removed/renamed exports.

**Minor observations (non-blocking):**
- The `fetchCode` function makes unauthenticated GitHub API requests, which are rate-limited to 60/hour. This is fine for a playground demo but worth noting in docs if it ever becomes a core feature.
- The gist regex uses `[a-f0-9]+` which is correct for gist IDs.
- Missing a `DOMImportConversion` (no `importDOM` static) — pasting an exported GitHub embed div won't reconstitute the node. Acceptable for a playground-only feature.

**CI:** CLA ✅, Vercel deploys ✅. Full test matrix not yet triggered (likely awaiting PR age/label). No blocking failures.

Ready to approve.

File: packages/lexical-playground/src/plugins/GitHubCodeExtension/index.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+export type GitHubCodePayload = {
+  url: string;
+  owner: string;
+  repo: string;
+  path: string;
+  branch: string;
+  startLine?: number;
+  endLine?: number;
+  language?: string;
+};
```
Comment (etrepum) [inline packages/lexical-playground/src/plugins/GitHubCodeExtension/index.ts:28]:
```suggestion
export interface GitHubCodePayload {
  url: string;
  owner: string;
  repo: string;
  path: string;
  branch: string;
  startLine?: number;
  endLine?: number;
  language?: string;
}
```

File: packages/lexical-playground/src/nodes/GitHubCodeNode.tsx [hunk 2]
```diff
@@ -0,0 +1,502 @@
...
+    path: string;
+    branch: string;
+    startLine?: number;
+    endLine?: number;
+    language?: string;
+  },
+  SerializedDecoratorBlockNode
+>;
+
+export class GitHubCodeNode extends DecoratorBlockNode {
```
Comment (etrepum) [inline packages/lexical-playground/src/nodes/GitHubCodeNode.tsx:348]: Using $config and NodeState will make a lot of this boilerplate go away

File: packages/lexical-playground/src/nodes/PlaygroundNodes.ts [hunk 3]
```diff
@@ -64,6 +65,7 @@ const PlaygroundNodes: Klass<LexicalNode>[] = [
   TweetNode,
   YouTubeNode,
   FigmaNode,
+  GitHubCodeNode,
```
Comment (etrepum) [inline packages/lexical-playground/src/nodes/PlaygroundNodes.ts:68]: This is a legacy file that you don't need to update, the extension takes care of configuring the node

**Review by etrepum — COMMENTED** [review]
I haven't done a close review of the react code yet but the line numbers aren't set up properly, they overflow and scroll separately from the content

Image

This PR's title and description also doesn't follow the [pull request template](https://raw.githubusercontent.com/facebook/lexical/refs/heads/main/.github/pull_request_template.md)

**Comment by etrepum** [comment]
@JiatangDong Why do you keep opening and closing PRs here? It's a waste of maintainer time if you don't intend for them to be reviewed and merged. This is the fourth time.
